crypto/crypto.go

```diff
@@ -27,18 +25,19 @@ func Digest(ms ...[]byte) []byte {
 	return ret
 }
 
-func GenerateKey() KeyPair {
-	pk, sk, _ := ed25519.GenerateKey(rand.Reader)
-	return KeyPair{
-		PrivateKey: sk,
-		PublicKey:  pk,
-	}
+func GenerateKey() (SigningKey, error) {
+	_, sk, err := ed25519.GenerateKey(rand.Reader)
+	return SigningKey(sk), err
 }
 
-func Sign(key KeyPair, message []byte) []byte {
-	return ed25519.Sign(key.PrivateKey, message)
+func Sign(key SigningKey, message []byte) []byte {
+	return ed25519.Sign(ed25519.PrivateKey(key), message)
 }
 
-func Verify(key KeyPair, message, sig []byte) bool {
-	return ed25519.Verify(key.PublicKey, message, sig)
+func Verify(key SigningKey, message, sig []byte) bool {
+	pk, ok := ed25519.PrivateKey(key).Public().(ed25519.PublicKey)
+	if !ok {
+		return false
+	}
+	return ed25519.Verify(pk, message, sig)
```

Comment:
OK, I see. This Verify function is just a wrapper around `ed25519.Verify` which everyone who only has the public key would use.